packages/auth-remix/src/server.ts

```diff
@@ -0,0 +1,795 @@
+              core.createPKCESession()
+            );
+            return new Response(null, {
+              status: 302,
+              headers: {
+                "Set-Cookie": `${this.options.pkceVerifierCookieName}=${pkceSession.verifier}; HttpOnly`,
+                Location:
+                  path.split("/").pop() === "signup"
+                    ? pkceSession.getHostedUISignupUrl()
+                    : pkceSession.getHostedUISigninUrl(),
+              },
+            });
+          }
+          case "emailpassword/verify": {
+            if (!onEmailVerify) {
+              throw new Error(
+                `'onEmailVerify' auth route handler not configured`
+              );
+            }
+            const verificationToken = searchParams.get("verification_token");
+            const verifier =
+              parseCookies(request)[this.options.pkceVerifierCookieName];
+
+            if (!verificationToken) {
+              return onEmailVerify({
+                error: new Error("no verification_token in response"),
+              });
+            }
+
+            if (!verifier) {
+              return onEmailVerify({
+                error: new Error("no pkce verifier cookie found"),
+                verificationToken,
+              });
+            }
+
+            let tokenData: TokenData;
+
+            try {
+              tokenData = await (
+                await this.core
+              ).verifyEmailPasswordSignup(verificationToken, verifier);
+            } catch (err) {
+              return onEmailVerify({
+                error: err instanceof Error ? err : new Error(String(err)),
+                verificationToken,
+              });
+            }
+
+            const headers = new Headers();
+            headers.append(
+              "Set-Cookie",
+              `${this.options.authCookieName}=${tokenData.auth_token}; HttpOnly; SameSite=strict; Path=/`
+            );
+
+            return oathCallbackCall(onEmailVerify, headers, {
+              error: null,
+              tokenData,
+            });
+          }
+
+          case "signout": {
+            if (!onSignout) {
+              throw new Error(`'onSignout' auth route handler not configured`);
+            }
+
+            const headers = new Headers({
+              "Set-Cookie": `${
+                this.options.authCookieName
+              }=; HttpOnly; SameSite=strict; Expires=${Date.now()}; Path=/`,
+            });
+
+            return oathCallbackCall(onSignout, headers);
+          }
+
+          default:
+            throw new Error("Unknown auth route");
+        }
+      },
+    };
+  }
+
+  async emailPasswordSignUp(
+    req: Request,
+    data?: { email: string; password: string }
+  ): Promise<{ tokenData: TokenData; headers: Headers }>;
+  async emailPasswordSignUp<Res extends Response>(
+    req: Request,
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
```

Comment:
Since we're following remix's behaviour, where if you return something that isn't a response, we wrap it in `json()`, these overloads need to be updated to:

```suggestion
  async emailPasswordSignUp<Res>(
    req: Request,
    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
  ): Promise<Res extends Response ? Res : TypedResponse<Res>>;
```